src/service/rooms/pdu_metadata/mod.rs

```diff
@@ -121,70 +117,30 @@ impl Service {
                     .take_while(|(k, _)| Some(k) != to.as_ref()) // Stop at `to`
 					.collect();
 
-				next_token = events_after.last().map(|(count, _)| count).copied();
+		let next_token = events.last().map(|(count, _)| count).copied();
 
-				let events_after: Vec<_> = events_after
+		let events_chunk: Vec<_> = match dir {
+			Direction::Forward => events
+				.into_iter()
+				.map(|(_, pdu)| pdu.to_message_like_event())
+				.collect(),
+			Direction::Backward => events
 					.into_iter()
 					.rev() // relations are always most recent first
 					.map(|(_, pdu)| pdu.to_message_like_event())
-					.collect();
-
-				Ok(get_relating_events::v1::Response {
-					chunk: events_after,
-					next_batch: next_token.map(|t| t.stringify()),
-					prev_batch: Some(from.stringify()),
-					recursion_depth: if recurse {
-						Some(depth.into())
-					} else {
-						None
-					},
-				})
-			},
-			Direction::Backward => {
-				let relations_until = &self.relations_until(sender_user, room_id, target, from, depth)?;
-				let events_before: Vec<_> = relations_until
-                    .iter()
-                    .filter(|(_, pdu)| {
-							filter_event_type.as_ref().map_or(true, |t| &pdu.kind == t)
-								&& if let Ok(content) =
-                                serde_json::from_str::<ExtractRelatesToEventId>(pdu.content.get())
-								{
-									filter_rel_type
-										.as_ref()
-										.map_or(true, |r| &content.relates_to.rel_type == r)
-								} else {
-									false
-								}
-						})
-					.take(limit)
-					.filter(|(_, pdu)| {
-						self.services
-							.state_accessor
-							.user_can_see_event(sender_user, room_id, &pdu.event_id)
```

Comment:
Does `user_can_see_event` get called anywhere else? This feels like a pretty important thing to have.

Reply:
Oops nevermind, just expanded the code above.